### DISCUSSION TITLE
heavy calculations inside sycamore::render closure

### DISCUSSION THREAD

**Discussion by ghost** [sycamore-rs/sycamore#514] [Q&A] [answered]
Hi!
Does `sycamore::render` closure run once, or can it be restarted occasionally?
The fact is that this is the only place (except for components) where there is `cx` And there it is convenient to prepare data, which is then transferred to `App`'s properties.
Or am I doing something wrong?

**Comment by lukechu10** [comment, marked as answer]
`sycamore::render` only runs once. It essentially works exactly the same as a component. I don't understand why you would want the closure to run multiple times. Usually, you should try to put all your logic in your `App` component to keep `fn main` as simple as possible.

If you have reactive data, you could use `create_memo` or `create_selector` etc...
